src/test_application/activescenes/scene_vehicles.cpp

```diff
@@ -187,44 +258,102 @@ Session setup_mach_rocket(
 
     top_get< MachTypeToEvt_t >(topData, idMachEvtTags).at(gc_mtMagicRocket) = tgMhRocketEvt;
 
-    // TODO: This session is not needed? Architecture prefers fine grained
-    //       individual sessions: thrust, plume FX, sounds, etc...
+    machRocket.task() = rBuilder.task().assign({tgSceneEvt, tgLinkReq, tgLinkMhUpdMod}).data(
+            "Allocate Machine update bitset for MagicRocket",
+            TopDataIds_t{idScnParts, idUpdMach},
+            gen_allocate_mach_bitsets<gc_mtMagicRocket>());
+
+    return machRocket;
+}
+
+Session setup_mach_rcsdriver(
+        Builder_t&                  rBuilder,
+        ArrayView<entt::any> const  topData,
+        Tags&                       rTags,
+        Session const&              scnCommon,
+        Session const&              parts,
+        Session const&              signalsFloat)
+{
+    OSP_SESSION_UNPACK_TAGS(scnCommon,      TESTAPP_COMMON_SCENE);
+    OSP_SESSION_UNPACK_DATA(signalsFloat,   TESTAPP_SIGNALS_FLOAT)
+    OSP_SESSION_UNPACK_TAGS(signalsFloat,   TESTAPP_SIGNALS_FLOAT);
+    OSP_SESSION_UNPACK_DATA(parts,          TESTAPP_PARTS);
+    OSP_SESSION_UNPACK_TAGS(parts,          TESTAPP_PARTS);
+
+    using namespace adera;
+
+    Session machRcsDriver;
+    OSP_SESSION_ACQUIRE_TAGS(machRcsDriver, rTags,   TESTAPP_MACH_RCSDRIVER);
 
-    machRocket.task() = rBuilder.task().assign({tgMhRocketEvt, tgSigFloatUpdReq, tgSigFloatValReq}).data(
-            "MagicRockets print throttle input values",
-            TopDataIds_t{           idScnParts,                      idUpdMach,                       idSigValFloat },
-            wrap_args([] (ACtxParts& rScnParts, UpdMachPerType const& rUpdMach, SignalValues_t<float>& rSigValFloat) noexcept
+    top_get< MachTypeToEvt_t >(topData, idMachEvtTags).at(gc_mtRcsDriver) = tgMhRcsDriverEvt;
+
+    machRcsDriver.task() = rBuilder.task().assign({tgMhRcsDriverEvt, tgSigFloatUpdMod}).data(
+            "RCS Drivers calculate new values",
+            TopDataIds_t{           idScnParts,                      idUpdMach,                       idSigValFloat,                    idSigUpdFloat },
+            wrap_args([] (ACtxParts& rScnParts, UpdMachPerType const& rUpdMach, SignalValues_t<float>& rSigValFloat, UpdateNodes<float>& rSigUpdFloat) noexcept
     {
         Nodes const &rFloatNodes = rScnParts.m_nodePerType[gc_ntSigFloat];
-        PerMachType &rRockets = rScnParts.m_machines.m_perType[gc_mtMagicRocket];
+        PerMachType &rRockets = rScnParts.m_machines.m_perType[gc_mtRcsDriver];
 
-        for (MachLocalId const local : rUpdMach.m_localDirty[gc_mtMagicRocket].ones())
+        for (MachLocalId const local : rUpdMach.m_localDirty[gc_mtRcsDriver].ones())
```

Comment:
RCS driver code starts here. This loop iterates all RCS drivers currently marked dirty. It reads all 12 connected floats, then uses z-adam's thruster_influence function to calculate a throttle output.